### PR TITLE
News feed relative images

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #7805 .

#### Summary of Changes
Displaying RSS feeds using com_newsfeed, many providers mistakenly use relative addressing to display images which results in feed display which includes numerous broken image links.

This code is provided by @monkeytrainer12

### CODE REVIEW

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php [hunk 1]
```diff
@@ -131,6 +131,9 @@
 						$text = JFilterOutput::stripImages($text);
 					}
 					$text = JHtml::_('string.truncate', $text, $this->params->get('feed_character_count'));
+					$domain = preg_replace("~^(https?://[^/]).$~", "\1/", $this->rssDoc[ $i ]->uri);
```
Comment (jeckodevelopment) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Is correct to always use "HTTPS" ?

Comment (jeckodevelopment) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Are you sure that we should use always "HTTPS"?

**Comment by hardiktailored** [comment]
I checked the code, I can see slash removed when URL starting like this src="/", but I think you have not included match for URL like src="../".
Can't we forbid to use relative paths and force to use absolute path only?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11414">issues.joomla.org/joomla-cms/11414</a>.</sub>

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php (hunk 1 above)
Comment (brianteeman) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: nope thats wrong - i will update

**Comment by brianteeman** [comment]
@hardiktailored 
> Can't we forbid to use relative paths and force to use absolute path only?
No because the newsfeed is NOT from our web site so we cant force anything

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php (hunk 1 above)
Comment (jeckodevelopment) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: thank you @brianteeman

Comment (brianteeman) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Sorry - I need to drink more coffee before replying
https? will correctly match both http and https 
you can see that at regexpal.com

**Comment by hardiktailored** [comment]
@brianteeman Yes, but can we append http:// or https:// in src path when not starting with one of them like replacing src="/" with src="http://"?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11414">issues.joomla.org/joomla-cms/11414</a>.</sub>

**Comment by hardiktailored** [comment]
Currently, News Feed is not loading any images. I am testing on 3.6.2.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11414">issues.joomla.org/joomla-cms/11414</a>.</sub>

**Comment by brianteeman** [comment]
@hardiktailored what newsfeed are you trying<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11414">issues.joomla.org/joomla-cms/11414</a>.</sub>

**Comment by brianteeman** [comment]
So long and farewell

**Pull request closed without merging**